mkdocs_awesome_nav/nav/directory.py

```diff
@@ -62,7 +85,9 @@ def _create_section(self, context: MkdocsFilesContext, *, section_type: Type[Nav
 
         resolved_children: list[NavPage | NavSection | NavLink] = resolve_in_priority_order(parsed_children, context)
 
-        return section_type(resolved_children, path=self.path, title=self._title)
+        title = self._resolved_title or dirname_to_title(self.path.name)
```

Comment:
Calling `dirname_to_title` shouldn't be necessary here as it's already called by `_generate_title`

Reply:
The dirname_to_title() is necessary as a fallback for the RootNavDirectory case where _resolved_title is never set because RootNavDirectory.resolve() bypasses the parent's resolve() method and calls _create_section() directly.